### PR TITLE
lops2019 integration

### LINKED ISSUE #1224: LOPS2019-integraatio
Toteutuksesta jätetään tässä vaiheessa yhdistelmäopintojaksot pois.

### PR DESCRIPTION
**Pull request by anttiviljakainen**
Closes #1224

### CODE REVIEW

File: persistence/src/main/java/fi/otavanopisto/pyramus/domainmodel/koski/KoskiPersonState.java [hunk 1]
```diff
@@ -19,15 +19,17 @@
   MISSING_STUDYAPPROVEDATE,
   MISSING_STUDYENDREASONMAPPING,
   NO_STUDYSTARTDATE,
-  EXISTING_INTERNETIX_STUDYPERMIT_WITHOUT_CREDITS;
+  EXISTING_INTERNETIX_STUDYPERMIT_WITHOUT_CREDITS,
+  CONFLICTING_COURSELENGHT,
```
Comment (anttiviljakainen) [inline persistence/src/main/java/fi/otavanopisto/pyramus/domainmodel/koski/KoskiPersonState.java:23]: Typo

File: framework/src/main/java/fi/otavanopisto/pyramus/koski/StudentSubjectSelections.java [hunk 2]
```diff
@@ -3,11 +3,23 @@
...
+import fi.otavanopisto.pyramus.koski.koodisto.KoskiOppiaineetYleissivistava;
+
 public class StudentSubjectSelections {
 
   public StudentSubjectSelections() {
   }
 
+  public KoskiOppiaineetYleissivistava koskiKoodi(String languageCode) {
+    return
+        isALanguage(languageCode) ? KoskiOppiaineetYleissivistava.A1 :
```
Comment (pkukkon) [inline framework/src/main/java/fi/otavanopisto/pyramus/koski/StudentSubjectSelections.java:15]: A and A1 both map to KoskiOppiaineetYleissivistava.A1?

File: framework/src/main/java/fi/otavanopisto/pyramus/koski/model/lukio/ops2019/LukionOppiaineenOppimaaranKoulutusmoduuli2019.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  public LukionOppiaineidenOppimaaratKoodi2019 getTunniste() {
+    return tunniste;
+  }
+
+  private final LukionOppiaineidenOppimaaratKoodi2019 tunniste = new LukionOppiaineidenOppimaaratKoodi2019();
+  private String perusteenDiaarinumero;
+  
+  public class LukionOppiaineidenOppimaaratKoodi2019 {
+    public final String koodiarvo = "lukionaineopinnot";
```
Comment (pkukkon) [inline framework/src/main/java/fi/otavanopisto/pyramus/koski/model/lukio/ops2019/LukionOppiaineenOppimaaranKoulutusmoduuli2019.java:21]: SuorituksenTyyppi.lukionaineopinnot?

File: framework/src/main/java/fi/otavanopisto/pyramus/koski/model/lukio/ops2019/LukionOppiaineenSuoritus2019.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  public boolean isSuoritettuErityisenäTutkintona() {
+    return suoritettuErityisenäTutkintona;
+  }
+
+  public void setSuoritettuErityisenäTutkintona(boolean suoritettuErityisenäTutkintona) {
+    this.suoritettuErityisenäTutkintona = suoritettuErityisenäTutkintona;
+  }
+
+  private boolean suoritettuErityisenäTutkintona;
```
Comment (pkukkon) [inline framework/src/main/java/fi/otavanopisto/pyramus/koski/model/lukio/ops2019/LukionOppiaineenSuoritus2019.java:53]: suoritettuErityisenaTutkintona? (mere aesthetics; scandinavian letters in variable names seem to mess up github code highlighting)

File: framework/src/main/java/fi/otavanopisto/pyramus/koski/model/lukio/ops2019/LukionOppiaineenOppimaaranKoulutusmoduuli2019.java (hunk 3 above)
Comment (anttiviljakainen) [reply framework/src/main/java/fi/otavanopisto/pyramus/koski/model/lukio/ops2019/LukionOppiaineenOppimaaranKoulutusmoduuli2019.java:21]: Valitettavasti eri asiasta kyse [https://koski.opintopolku.fi/koski/dokumentaatio/koski-oppija-schema.html?entity=lukionopiskeluoikeus#lukionoppiaineidenoppim%C3%A4%C3%A4r%C3%A4t2019](url)

File: framework/src/main/java/fi/otavanopisto/pyramus/koski/model/lukio/ops2019/LukionOppiaineenSuoritus2019.java (hunk 4 above)
Comment (anttiviljakainen) [reply framework/src/main/java/fi/otavanopisto/pyramus/koski/model/lukio/ops2019/LukionOppiaineenSuoritus2019.java:53]: Function over form :smiling_imp:

File: framework/src/main/java/fi/otavanopisto/pyramus/koski/StudentSubjectSelections.java (hunk 2 above)
Comment (anttiviljakainen) [reply framework/src/main/java/fi/otavanopisto/pyramus/koski/StudentSubjectSelections.java]: Yes - the mapping is weird and was also changed for 2019.

**Review by pkukkon — APPROVED** [review]

**Pull request merged**